src/CSET/operators/constraints.py

```diff
@@ -65,29 +66,138 @@ def generate_var_constraint(varname: str, **kwargs) -> iris.Constraint:
 
 
 def generate_model_level_constraint(
-    model_level_number: Union[int, str], **kwargs
+    model_level: int | list[int], **kwargs
 ) -> iris.Constraint:
     """Generate constraint for a particular model level number.
 
-    Operator that takes a CF compliant model_level_number string, and uses iris to
+    Operator that takes a CF compliant model_level_number int or list of int, and uses iris to
     generate a constraint to be passed into the read operator to minimize the
     CubeList the read operator loads and speed up loading.
 
     Arguments
     ---------
-    model_level_number: str
-        CF compliant model level number.
+    model_level: int|list
+        CF compliant model levels.
 
     Returns
     -------
-    model_level_number_constraint: iris.Constraint
+    model_level_constraint: iris.Constraint
     """
-    # Cast to int in case a string is given.
-    model_level_number = int(model_level_number)
-    model_level_number_constraint = iris.Constraint(
-        model_level_number=model_level_number
-    )
-    return model_level_number_constraint
+    # turn into a list in case is iterable
+    if not isinstance(model_level, Iterable):
+        model_level = [model_level]
+
+    if len(model_level) == 0:
+        # If none specified reject cubes with any model level coordinate.
+        def no_model_level_number(cube):
+            cube.coords()
+
+            try:
+                cube.coord("model_level_number")
+                # return false if found respective level coordinate
+                return False
+            except iris.exceptions.CoordinateNotFoundError:
+                # do nothing if not found coordinate
+                return True
```

Comment:
```suggestion
            # Check for coordinate existence.
            return bool(cube.coords("model_level_number"))
```
A bit of a nicer way to do this.